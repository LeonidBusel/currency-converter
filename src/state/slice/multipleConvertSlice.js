import { createSlice } from '@reduxjs/toolkit'

export const initialState = {
    isFetching: false,
    columns: [],
    dataSource: [],
    currency: null
}

export const multipleConvertSlice = createSlice({
    name: 'multipleConvert',
    initialState,
    reducers: {
        multipleConvertFetch: state => {
            state.isFetching = true;
        },
        multipleConvertFetchSuccess: (state, { payload }) => {
            state.isFetching = false;
            state.columns = payload.columns.slice();
            state.dataSource = payload.dataSource.slice();
            state.currency = payload.currency;
        },
        multipleConvertFetchFail: (state) => {
            state.isFetching = false;
            state.columns = [];
            state.dataSource = [];
            state.currency = null;
        },
        multipleConvertClear: (state) => {
            state.isFetching = false;
            state.columns = [];
            state.dataSource = [];
            state.currency = null;
        }
    }
})

export const { multipleConvertFetch, multipleConvertFetchSuccess, multipleConvertFetchFail, multipleConvertClear } = multipleConvertSlice.actions

export default multipleConvertSlice.reducer