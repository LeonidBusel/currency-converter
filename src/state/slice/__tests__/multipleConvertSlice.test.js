import multipleConvertSlice, {
    initialState,
    multipleConvertFetch,
    multipleConvertFetchSuccess,
    multipleConvertFetchFail,
    multipleConvertClear
} from "../multipleConvertSlice";

const modifyState = {
    "isFetching": false,
    "columns": [1, 2],
    "dataSource": [3, 4],
    "currency": 55
}

describe('multipleConvertSlice', () => {
    it('handle multipleConvertFetch correctly', () => {
        const actual = multipleConvertSlice(initialState, multipleConvertFetch());

        expect(actual).toMatchSnapshot();
    });

    it('handle multipleConvertFetchSuccess correctly', () => {
        const actual = multipleConvertSlice(initialState, multipleConvertFetchSuccess({ columns: [1, 2], dataSource: [3, 4], currency: 55 }));

        expect(actual).toMatchSnapshot();
    });

    it('handle multipleConvertFetchFail correctly', () => {
        const actual = multipleConvertSlice(modifyState, multipleConvertFetchFail());

        expect(actual).toMatchSnapshot();
    });

    it('handle multipleConvertClear correctly', () => {
        const actual = multipleConvertSlice(modifyState, multipleConvertClear());

        expect(actual).toMatchSnapshot();
    });
});