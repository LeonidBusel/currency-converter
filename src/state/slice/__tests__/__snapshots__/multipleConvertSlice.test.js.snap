// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`multipleConvertSlice handle multipleConvertClear correctly 1`] = `
Object {
  "columns": Array [],
  "currency": null,
  "dataSource": Array [],
  "isFetching": false,
}
`;

exports[`multipleConvertSlice handle multipleConvertFetch correctly 1`] = `
Object {
  "columns": Array [],
  "currency": null,
  "dataSource": Array [],
  "isFetching": true,
}
`;

exports[`multipleConvertSlice handle multipleConvertFetchFail correctly 1`] = `
Object {
  "columns": Array [],
  "currency": null,
  "dataSource": Array [],
  "isFetching": false,
}
`;

exports[`multipleConvertSlice handle multipleConvertFetchSuccess correctly 1`] = `
Object {
  "columns": Array [
    1,
    2,
  ],
  "currency": 55,
  "dataSource": Array [
    3,
    4,
  ],
  "isFetching": false,
}
`;
